test/ProgressOnderwijsUtils.Tests/ApprovalTestTest.cs

```diff
@@ -57,4 +57,34 @@ public void ApprovalWhenFileIsSameDoesNotThrow()
             () => File.Delete(approval.ApprovalPath)
         );
     }
+
+    [Fact]
+    public void Approval_failure_gives_correct_diff()
+    {
+        var approval = ApprovalTest.CreateHere();
+        File.WriteAllText(approval.ApprovalPath, "Hello world!");
+        Utils.TryWithCleanup(
+            () => {
+                var changed = approval.IsChangedFrom("Hello astronauts ...", out var diff);
+                PAssert.That(() => changed);
+                PAssert.That(() => diff == "'world!' »» 'astron'");
+            },
+            () => File.Delete(approval.ApprovalPath)
+        );
```

Comment:
this test might be a little simpler if we split the file-system logic from the given-FS-state-diff-computation logic.

Reply:
(but fine as is)